tools/clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -277,15 +278,17 @@ namespace {
           if (it->first->isValid() && !it->second->IsSystemFile) {
             // If main file, write that to metadata first.
             // Add the rest to filesMap to sort by name.
-            llvm::SmallString<128> NormalizedPath;
-            llvm::sys::path::native(it->first->getName(), NormalizedPath);
             if (CodeGenOpts.MainFileName.compare(it->first->getName()) == 0) {
               assert(!bFoundMainFile && "otherwise, more than one file matches main filename");
-              AddFile(NormalizedPath, it->second->getRawBuffer()->getBuffer());
+              AddFile(hlsl::NormalizePath(it->first->getName()),
+                      it->second->getRawBuffer()->getBuffer());
               bFoundMainFile = true;
             } else {
-              filesMap[NormalizedPath.str()] =
-                  it->second->getRawBuffer()->getBuffer();
+              // We want the include file paths to match the values passed into
+              // the include handlers exactly. The SourceManager entries should
+              // match it except the call to MakeAbsoluteOrCurDirRelative.
+              std::string path = hlsl::NormalizePath(it->first->getName());
```

Comment:
Consider declaring this variable before the if condition, and then using it in both the if and the else clause.

Reply:
Done.